pkg/kopia/cli/internal/log/log.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2024 The Kanister Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"context"
+	"io"
+
+	"github.com/kanisterio/kanister/pkg/field"
+	"github.com/kanisterio/kanister/pkg/log"
+)
+
+// NopLogger is a logger that does nothing.
```

Comment:
Why do we need this?

Reply:
This Noop logger is used inside [s3 storage](https://github.com/kanisterio/kanister/blob/b75596634b348d3c3013a6dcef341f62a543b16a/pkg/kopia/cli/repository/storage/s3/s3.go#L30-L32) as a default logger if no logger is provided. 
It helps to avoid code cluttering and remove some code bumps. 

You don't need to write something like below: 

```
    if logger != nil {
        logger.Print("Removing trailing slashes from the endpoint")
    }
```
and can simply use:

```
    logger.Print("Removing trailing slashes from the endpoint")
```